### PR TITLE
Edit variations: Inventory

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds the inventory editing functionality to Variation details. The current Inventory screen functionality has been extracted from `ProductDetailViewModel` into a separate one so that it could be reused for variations.

image

**To test:**
1. Go to Products -> Tap on a product
2. Tap on the Inventory property
3. Verify the inventory data is shown as expected
4. Change some value
5. Notice the Done button appears
6. Tap on the back button
7. Notice that a discard dialog is shown
8. Tap Cancel
9. Tap on the Done button
10. Verify that the inventory information was updated in product details
11. Tap on Update
12. Verify that the product data was updated
13. Go back to Products -> Tap on a variable product -> Tap on Variations -> Tap on a variation item
14. Tap on the Inventory property
15. Notice that the "Limit one per order" switch is _not_ shown
16. Repeat steps 3-12 for the variation

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailCardBuilder.kt [hunk 1]
```diff
@@ -349,9 +354,20 @@ class ProductDetailCardBuilder(
             R.drawable.ic_gridicons_list_checkmark,
             true
         ) {
+            PRODUCT_DETAIL_VIEW_INVENTORY_SETTINGS_TAPPED
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailCardBuilder.kt]: np: not sure if this needed here?

**Review by anitaa1990 — COMMENTED** [review]
Great work @0nko 🥇 ! I've left one nit pick comment in the code and a few notes after testing.

- Probably not directly related to the changes in this PR but I noticed that the description alignment in the below screenshot looks a bit off.



- The `Manage Stock` switch does not retain it's value after a configuration change, although the SKU field does. Actually, on testing again, I noticed that none of the switch values are retained on config changes but the editText values are retained so, for instance, if I enable manage stock and enter some quantity, the quantity is updated but the switch is not. 



- The same thing seems to be happening when scheduling a sale for Simple and variable products.





- For some reason, the `stockQuantity` is reset to 0, after the variation is successfully updated.

**Comment by 0nko** [comment]
Thanks for the review @anitaa1990. All issues are fixed except for "For some reason, the stockQuantity is reset to 0, after the variation is successfully updated.". I couldn't reproduce this. Can you please check if it's still happening and if it is, how to reproduce it? Thanks!

**Review by anitaa1990 — APPROVED** [review]
Thanks @0nko! I'm not actually able to reproduce the issue with the stock quantity anymore 😕  
LGTM :shipit:

**Pull request merged**